.pre-commit-hooks.yaml

```diff
@@ -40,6 +40,11 @@
     language: python
     types: [text, executable]
     stages: [commit, push, manual]
+-   id: check-illegal-windows-names
+    name: check illegal windows names
+    entry: Illegal windows filenames detected
+    language: fail
+    files: '(?i)(^|/)(CON|PRN|AUX|NUL|COM[1-9]|LPT[1-9])(\.|/|$)'
```

Comment:
can we add a quick little test which loads this regex from the configuration and runs it against a few expected values?

Reply:
@asottile @ericfrederich I recommend checking for the `:` character as well (see my comment in #589 regarding :Zone.Identifier files).

Reply:
@jleclanche looks like this got merged without code checking for the `:` but no release has been made yet.

I just checked and it's definitely a problem on Windows.  I created a file `foo:bar` on linux, commited, pushed, then pulled down the repo from Windows and got greeted with `error: invalid path 'foo:bar'`

I'll add a test for it and a new PR